### ISSUE TITLE
Wrong network ip address for the frontend (Mac M1)

### ISSUE THREAD

**Issue by xdatap1** [pieroit/cheshire-cat#50]
When I start the container, it shows:

frontend  |   ➜  Local:   http://localhost:3000/
frontend  |   ➜  Network: http://172.19.0.3:3000/

But the IP address should be instead 192.168.1.4

Mac M1 - OSX 13.2.1

**Comment by pieroit** [comment]
Does the "wrong" url work?

**Comment by xdatap1** [comment]
Nope, it doesn't work. I tested it from the same machine and another device  in the same network and this address is not reachable. I tried 192.168.1.4:3000 from another device and the html loads properly but it shows "Something went wrong while connecting to the server. Please try again later". Using localhost the page loads properly.

**Comment by pieroit** [comment]
I don't know how to fix at the moment.
Leaving the issue open in case someone has suggestions.

Thank you for reaching out, I hope you have fun with the cat!

**Comment by vcapoccia** [comment]
Same issue for me on Linux Debian 11 minimal install.
I'll try with with a whole desktop vm install, using the cat from localhost.

![image](https://user-images.githubusercontent.com/1284688/228665374-522429be-8447-43e9-8c04-2d67df2f517d.png)

**Comment by vcapoccia** [comment]
![therabbithole](https://user-images.githubusercontent.com/1284688/228671138-e3201c2e-f79d-444a-b39a-4512d3679a16.PNG)

Seems perfectly. I take this opportunity to congratulate you, Piero, because you are a clear example of Italian creativity and genius.
Looking forward to see other fireworks from there!

**Comment by pieroit** [comment]
thank you a lot  @vcapoccia !!!
Let's hack the cat! ^._.^

**Comment by antonioru** [comment]
@xdatap1 @vcapoccia 

We're using [VITE](https://vitejs.dev/) to run the front-end in 'development' mode, so the localhost URL is the correct one. 
VITE will also try to expose the network IP, but since it depends on the machine configuration, it might not work if the machine network configuration has been customized. 
Ofc we cannot store machine-specific customisation in the repository, so my suggestion for the moment would be to just ignore that message or feel free to open a PR to hide the network address ;)

**Comment by vcapoccia** [comment]
@antonioru first of all, thank you for the reply. I don't have any custom network configuration (such for example static ip). Even if you cannot store machine-specific customization in the repo, maybe some specific instruction for network configuration helps people to spread the cat's verb worldwide! :)
In my case I just installed a Debian vm, apt installed docker, vim, git. Then followed the (really simple) md

**Comment by xdatap1** [comment]
@antonioru I am on a Mac with IP from the DHCP, nothing strange. Regardless of the IP shown in the loading message, the front end is reachable from the machine IP though, but from accessing it from other addresses, you get "Something went wrong while connecting to the server. Please try again later" so the problem is FE-BE communication when loading the FE with the machine's IP. If you want to do some debugging together, we could set up a Zoom call, and I'll show it to you.

**Comment by marco-cagnoli** [comment]
I'm experiencing the same problem.
My coniguration: 
- client Windows (Browser)
- server linux
- I'm able to connect to port 3000
- what is not working is the web socket on port 1865
- I changed in frontend/src/config.ts ip address fro localhost to my remote host IP

Something strange: I tried to implement a simple server socket on my remote host (I wanted to be sure that my firewall was properly configuredy) and I had at the beginning several problems, exactly same problems I'm experiencing with tha cat. Then I used a specific python package (https://pypi.org/project/simple-websocket-server/) and the connection worked fine (on port 1865).

So I'm able to reach the remote host estabilishing a working web socket connection.

Now, the question is why I'm not able to connect to the cat.

May be some mapping port mapping in the docker-config.yml file?

Thanks

**Comment by antonioru** [comment]
> @antonioru I am on a Mac with IP from the DHCP, nothing strange. Regardless of the IP shown in the loading message, the front end is reachable from the machine IP though, but when accessing it from other addresses, you get "Something went wrong while connecting to the server. Please try again later" so the problem is FE-BE communication when loading the FE with the machine's IP. If you want to do some debugging together, we could set up a Zoom call, and I'll show it to you.

Hey, @xdatap1 thanks for looking into it. No need for a Zoom call the issue is quite clear to me. 
The reason why it happens is that the FE assumes the backend is always "localhost", so if you reach the FE from a different machine and it doesn't run the backend locally it won't be able to connect 

As suggested by @marco-cagnoli change the config in the FE folder and restart VITE, it should fix it for now

**Comment by marco-cagnoli** [comment]
I've just updated from git and now, changing the config.ts IPs, everything is working as expected!!!

**Comment by pieroit** [comment]
There is now a `.env.example` with some documentation. Both containers host and port can be customized.
Ignore vite logs in the console, as they are referring to the docker container ports and host.

**Comment by pribadihcr** [comment]
> Same issue for me on Linux Debian 11 minimal install. I'll try with with a whole desktop vm install, using the cat from localhost.
> 
> ![image](https://user-images.githubusercontent.com/1284688/228665374-522429be-8447-43e9-8c04-2d67df2f517d.png)

I got the same error. did you resolve it?

**Comment by vcapoccia** [comment]
I tryied with a config workaround, using a ssh localhost tunnel from client to Cat's server, and it work for me.

**Comment by pieroit** [comment]
Are you using the `.env` config?

https://cheshire-cat-ai.github.io/docs/administrators/env-variables/

Also
https://cheshire-cat-ai.github.io/docs/faq/
